master/internal/resourcemanagers/priority.go

```diff
@@ -185,7 +185,8 @@ func trySchedulingTaskViaPreemption(
 	log.Debugf("trying to schedule task %s by preempting other tasks", allocationRequest.Name)
 
 	for priority := model.MaxUserSchedulingPriority; priority > allocationPriority; priority-- {
-		for _, preemptionCandidate := range priorityToScheduledTaskMap[priority] {
+		for i := len(priorityToScheduledTaskMap[priority]) - 1; i >= 0; i-- {
```

Comment:
idea here is we should preempt newest jobs first when trying to make room for high priority jobs?

Reply:
yes an extension of "we prioritize jobs that have been submitted the longest" which I think was also the original intent?

Reply:
makes sense.